### PR TITLE
feat: supports validate field without @Valid

### PR DESCRIPTION
**Pull request by foreleven**

### CODE REVIEW

File: src/core/FormField.ts [hunk 1]
```diff
@@ -5,13 +5,16 @@ import get from 'lodash.get';
...
+  readonly field: string | symbol;
+
+  private readonly _schema?: yup.Schema<Values>;
 
   private _errors: FormError<Values> = {};
 
   private _touched: { [K in keyof Values]?: FormTouched<Values[K]> } = {};
 
-  constructor(schema: yup.Schema<Values>) {
+  constructor(field: string | symbol, schema?: yup.Schema<Values>) {
```
Comment (JounQin) [inline src/core/FormField.ts]:
```suggestion
  constructor(public readonly field: string | symbol, private readonly _schema?: yup.Schema<Values>) {
```

No need to declare and initial fields twice.

File: src/core/FormModel.ts [hunk 2]
```diff
@@ -14,24 +14,30 @@ export class FormModel<Values> {
     [K in keyof Values]: FormField<Values[K]>;
   } = {} as { [K in keyof Values]: FormField<Values[K]> };
 
-  setFormField<T extends keyof Values>(
+  setFormField<T extends keyof Values & (string | symbol)>(
```
Comment (JounQin) [inline src/core/FormModel.ts]: It strange and it is same as following:

```suggestion
  setFormField<T extends string | symbol>(
```

File: src/core/FormModel.ts [hunk 3]
```diff
@@ -14,24 +14,30 @@ export class FormModel<Values> {
...
-    this[fields][field] = new FormField(schema);
+    this[fields][field] = new FormField(field, schema);
   }
 
   getFormField<T extends keyof Values>(field: T): FormField<Values[T]> {
     return this[fields][field];
   }
 
-  validate<T extends keyof Values>(field: T): Promise<boolean> {
+  validate<T extends keyof Values & string>(
```
Comment (JounQin) [inline src/core/FormModel.ts]: As like as above.

```suggestion
  validate<T extends string>(
```

File: src/core/FormModel.ts (hunk 2 above)
Comment (foreleven) [reply src/core/FormModel.ts]: `setFormField<T extends string | symbol>` will miss the code suggestion for the key of `Values`

File: src/core/FormField.ts [hunk 4]
```diff
@@ -5,14 +5,16 @@ import get from 'lodash.get';
...
 
   private _touched: { [K in keyof Values]?: FormTouched<Values[K]> } = {};
 
-  constructor(schema: yup.Schema<Values>) {
-    this._schema = schema;
+  constructor(
+    public readonly field: string | symbol,
+    private readonly _schema?: yup.Schema<Values>,
+  ) {
+    this.field = field;
```
Comment (JounQin) [inline src/core/FormField.ts]: No need to do this.

**Pull request merged**